# Enable shared library build
# Copyright (C) 2009,2012 Georgi D. Sotirov <gdsotirov@dir.bg>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
#
# Visit SlackPack at http://sotirov-bg.net/slackpack/
#
# $Id: lapack-3.4.0.diff,v 1.1 2012/03/04 10:53:25 gsotirov Exp $
#

diff -urNad lapack-3.4.0-orig/SRC/Makefile lapack-3.4.0/SRC/Makefile
--- lapack-3.4.0-orig/SRC/Makefile	2011-10-04 05:58:05.000000000 +0300
+++ lapack-3.4.0/SRC/Makefile	2012-03-04 12:11:45.000000000 +0200
@@ -398,11 +398,10 @@
 ALLXOBJ = $(SXLASRC) $(DXLASRC) $(CXLASRC) $(ZXLASRC)
 endif
 
-all: ../$(LAPACKLIB)
+all: $(LAPACKLIB)
 
-../$(LAPACKLIB): $(ALLOBJ) $(ALLXOBJ)
-	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ) $(ALLXOBJ)
-	$(RANLIB) $@
+$(LAPACKLIB): $(ALLOBJ) $(ALLXOBJ)
+	$(FORTRAN) -shared -Wl,-soname=$@ -o $@ $(ALLOBJ) $(ALLXOBJ) $(BLASLIB)
 
 single: $(SLASRC) $(DSLASRC) $(SXLASRC) $(SCLAUX) $(ALLAUX)
 	$(ARCH) $(ARCHFLAGS) ../$(LAPACKLIB) $(SLASRC) $(DSLASRC) \
@@ -444,7 +443,7 @@
 	@FRC=$(FRC)
 
 clean:
-	rm -f *.o
+	rm -f *.o *.so.*
 
 .f.o: 
 	$(FORTRAN) $(OPTS) -c $< -o $@
diff -urNad lapack-3.4.0-orig/make.inc.example lapack-3.4.0/make.inc.example
--- lapack-3.4.0-orig/make.inc.example	2011-11-11 23:08:04.000000000 +0200
+++ lapack-3.4.0/make.inc.example	2012-03-04 12:17:05.000000000 +0200
@@ -3,6 +3,11 @@
 #  LAPACK, Version 3.4.0                                           #
 #  November 2011                                                   #
 ####################################################################
+
+VER_MAJ = 3
+VER_MIN = 4
+VER_PATCH = 0
+
 #
 SHELL = /bin/sh
 #  
@@ -13,9 +18,9 @@
 #  desired load options for your machine.
 #
 FORTRAN  = gfortran 
-OPTS     = -O2
+OPTS     = $(FFLAGS) -O2 -mieee-fp -fPIC
 DRVOPTS  = $(OPTS)
-NOOPT    = -O0
+NOOPT    = $(FFLAGS) -O0 -mieee-fp -ffloat-store -fPIC
 LOADER   = gfortran
 LOADOPTS =
 #
@@ -53,6 +58,7 @@
 #  machine-specific, optimized BLAS library should be used whenever
 #  possible.)
 #
-BLASLIB      = ../../librefblas.a
-LAPACKLIB    = liblapack.a
+BLASLIB      = -lblas
+LAPACKLIB    = liblapack.so.$(VER_MAJ).$(VER_MIN).$(VER_PATCH)
 TMGLIB       = libtmglib.a
+
